src/linear_algebra/tensor_product_basis.lean

```diff
@@ -50,3 +52,27 @@ finite_dimensional.of_fintype_basis
   (basis.tensor_product (basis.of_vector_space K V) (basis.of_vector_space K W))
 
 end field
+
+section finiteness
+
+variables {R : Type*} {M : Type*} {N : Type*}
+variables [comm_semiring R] [add_comm_monoid M] [module R M] [add_comm_monoid N] [module R N]
+variables [module.finite R M] [module.finite R N]
+
+instance tensor_product.finite : module.finite R (M ⊗[R] N) :=
+{ out := begin
+  rcases _inst_6.out with ⟨sM, hM⟩,
```

Comment:
Try to avoid using `_inst_6` explicitly, it is unstable since we do not control its name. You can write `: ... := infer_instance` or something like that. Same below.